crates/l1tx/src/deposit/deposit_tx.rs

```diff
@@ -74,14 +75,24 @@ fn validate_deposit_signature(
 
     // Extract and validate input signature
     let input = tx.input[0].clone();
-    let sig_bytes = &input.witness[0];
-    let schnorr_sig = Signature::from_slice(sig_bytes.get(..64)?).unwrap();
+
+    // Check if witness is present.
+    if input.witness.is_empty() {
+        return None;
+    }
+    let sig_witness = &input.witness[0];
+    if sig_witness.len() < 64 {
+        return None;
+    }
+    let sig_bytes = &sig_witness[..64];
+    let schnorr_sig = Signature::from_slice(sig_bytes).unwrap();
```

Comment:
This still has the `0xff`*64 panic I mentioned.  The `.get` slicing was fine before, the issue is that you were indexing (instead of `.get()`) and calling `.unwrap()`.

Reply:
Also, please don't use magic numbers like `64`. You can use constants such as https://docs.rs/bitcoin/latest/bitcoin/key/constants/constant.SCHNORR_SIGNATURE_SIZE.html

Reply:
Fixed.